OneCommunityGlobal/HighestGoodNetworkApp
# Yiyun - complete the configuration

# Description
This PR is one one the ESLint PR series.
It's for the below purposes:
1. run Prettier as one of the ESLint rules
2. add Husky lint check as pre-commit check
3. update the GitHub action
4. make sure current repo is lint error-free 
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/5071040/82a5e279-2f0f-4df7-b7a4-d47457964d92)

## Related PRS (if any):
N/A

## Main changes explained:
- `.eslintignore`, `.prettierignore`: update the files ignored by ESLint and Prettier
- `.eslintrc.js`: update ESLint rules to be more strict
- `.github/workflows/node.js.yml` update the github workflow action
- `.husky/pre-commit`: add Husky lint check as the pre-commit check
- `package.json`: update the lint check commands
- `public/config.js`: removed cause no need anymore
- `Utils/index.js`: removed cause no need anymore
- other changes are done by running `npm run lint` as the first time

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as any user
4. verify no functionality break because of this PR

## Screenshots or videos of changes:
Below images show run the commands `npm run lint` and `npm run lint:fix`
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/5071040/3229db93-50e4-4feb-8d6e-f9f1895cb0f7)
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/5071040/25aaf180-0bb0-42d5-9513-980bc533b57d)
Below image shows when commit your changes, it will automatically check if your changes are lint error-free.
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/5071040/e1bf9ab8-72e1-4816-8c18-e765685281db)

## Note:
Ref: https://www.youtube.com/watch?v=ZXW6Jn6or1w
